aten/src/ATen/native/cuda/ScatterGatherKernel.cu

```diff
@@ -288,7 +289,41 @@ void scatter_fill_cuda_kernel(Tensor& self, int64_t dim, const Tensor& index, Sc
   );
 }
 
+template <bool is_complex_t>
+struct scatter_add_op {
```

Comment:
nit: I find it slightly more readable if templating for complex is done on enable_if_t rather than on bool template arg, like here https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/NumericUtils.h#L37-L47